### PR TITLE
Windows fixes (VIAME)

### PR DESCRIPTION
**Pull request by neal-siekierski**
This PR includes fixes to the `viame/master` branch for it to compile on Windows, specifically with Visual Studio 15. Perhaps some of this is also relevant for `master`.

These are largely straightforward changes, but I have a couple concerns:
- What's the canonical place in a `CMakeLists.txt` file to call `add_compile_options`?
- `KipQueryService` requires C++17 because of one use of `[[maybe_unused]]`. Is it worth bumping up the required standard for this?

@mwoehlke-kitware @mattdawkins

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -4,6 +4,13 @@ set(CMAKE_CXX_STANDARD 11)
 
 project(visGUI)
 
+if(MSVC AND (MSVC_VERSION GREATER_EQUAL 1914))
+  # qtExtensions uses the __cplusplus macro to test for C++11 support
```
Comment (mwoehlke-kitware) [inline CMakeLists.txt]: This should be fixed by kitware/qtextensions@a1ebde053ccf71f60f1e6928ec04fea1aa025a28; are you sure you need this?

File: Plugins/VvQueryService/KipQueryService/CMakeLists.txt [hunk 2]
```diff
@@ -43,6 +43,9 @@ vg_add_qt_plugin(${PROJECT_NAME}
   ${vvKipQueryService_UiSources}
 )
 
+set_property(TARGET ${PROJECT_NAME}
+  PROPERTY CXX_STANDARD 17)
```
Comment (mwoehlke-kitware) [inline Plugins/VvQueryService/KipQueryService/CMakeLists.txt]: Why is this needed?

File: Libraries/VtkVgCore/vtkVgPlotTimeline.cxx [hunk 3]
```diff
@@ -232,7 +232,7 @@ struct IdToPointXIter
     : Points(pts), Pointer(start)
     {}
 
-  double operator*()
+  double operator*() const
```
Comment (mwoehlke-kitware) [inline Libraries/VtkVgCore/vtkVgPlotTimeline.cxx]: (FYI, it looks like this was also fixed by ace613f79b6a52f0404a1fa7e6cac105e602bba4.)

File: CMakeLists.txt (hunk 1 above)
Comment (neal-siekierski) [reply CMakeLists.txt]: The version of qtextensions I'm using has that commit. In building VIAME, however, it doesn't seem to use that flag when building vivia.

File: Plugins/VvQueryService/KipQueryService/CMakeLists.txt (hunk 2 above)
Comment (neal-siekierski) [reply Plugins/VvQueryService/KipQueryService/CMakeLists.txt]: Because of the use of `[[maybe_unused]]` here:
https://github.com/neal-siekierski/vivia/blob/viame/dev/windows_fixes/Plugins/VvQueryService/KipQueryService/vvKipQuerySession.cxx#L462
Without it, VS complains that `/std:c++17` is required.

File: CMakeLists.txt (hunk 1 above)
Comment (mwoehlke-kitware) [reply CMakeLists.txt]: That's... concerning. Does your `qtExtensionsConfigTargets.cmake` have exported compile options for `qtExtensionsHeaders`? (AFAIK it is working for other people...)

File: Plugins/VvQueryService/KipQueryService/CMakeLists.txt (hunk 2 above)
Comment (mwoehlke-kitware) [reply Plugins/VvQueryService/KipQueryService/CMakeLists.txt]: Sigh...

Removing the `[[maybe_unused]]` and adding a `Q_UNUSED(pluginsLoaded)` might be better. AFAIK this code is notionally supposed to work on GCC 4.8 which only "mostly" supports C++11 (although also some C++14 bits).

File: CMakeLists.txt (hunk 1 above)
Comment (neal-siekierski) [reply CMakeLists.txt]: It does not. Perhaps the problem is here, with the usage of `ADDED_FLAG`?:
https://github.com/Kitware/qtextensions/blob/a911d919bc8f01c6f3af067dc95a654b2f27cf16/cmake/config.cmake#L88

That variable is set using `PARENT_SCOPE`, but not by `qte_add_cxx_flags`, but rather its callee.

Comment (mwoehlke-kitware) [reply CMakeLists.txt]: Hmm, yes, that looks plausible. Care to submit a patch to qtExtensions? (I wonder how no one else has noticed this...)

**Comment by neal-siekierski** [comment]
I've switched to using `Q_UNUSED` instead of requiring C++17, and I've removed the qtextensions workaround in anticipation of its [`fix-interface-requirements`](https://github.com/Kitware/qtextensions/compare/fix-interface-requirements) branch getting merged.
